BUILD_ASSERT in net_if.h causes clang-tidy to fail compilation

### Describe the bug

Running `clang-tidy` on a file including `zephyr/include/zephyr/net/net_if.h` causes `clang-tidy` to fail compilation of the file. `clang-tidy` hits this line `BUILD_ASSERT((enum net_if_oper_state)(-1) > 0 && NET_IF_OPER_UNKNOWN == 0);` near line 915 and spits out the following build error:

```
zephyr/include/zephyr/net/net_if.h:915:15: error: static assertion expression is not an integral constant expression [clang-diagnostic-error]
  915 |         BUILD_ASSERT((enum net_if_oper_state)(-1) > 0 && NET_IF_OPER_UNKNOWN == 0);
      |                      ^
zephyr/include/zephyr/toolchain/gcc.h:76:50: note: expanded from macro 'BUILD_ASSERT'
   76 | #define BUILD_ASSERT(EXPR, MSG...) static_assert(EXPR, "" MSG)
      |                                                  ^~~~
zephyr/include/zephyr/net/net_if.h:915:15: note: integer value -1 is outside the valid range of values [0, 7] for the enumeration type 'net_if_oper_state'
  915 |         BUILD_ASSERT((enum net_if_oper_state)(-1) > 0 && NET_IF_OPER_UNKNOWN == 0);
```

This stops me from running `clang-tidy` on my own code.

### Regression

- [ ] This is a regression.

### Steps to reproduce

Run clang-tidy on a file including net_if.h.

### Relevant log output

```shell

```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

Present on bb699ff2df249addb9473d0bbd2f6949f008161e
LLVM 21

### Additional Context

I'd be happy to make a fix if I could better understand what the -1 check is actually trying to get at. Is it trying to assert that the enum is unsigned?

--- a/include/zephyr/net/net_if.h
+++ b/include/zephyr/net/net_if.h
@@ -912,7 +912,6 @@ static inline enum net_if_oper_state net_if_oper_state_set(
 		return NET_IF_OPER_UNKNOWN;
 	}
 
-	BUILD_ASSERT((enum net_if_oper_state)(-1) > 0 && NET_IF_OPER_UNKNOWN == 0);
 	if (oper_state <= NET_IF_OPER_UP) {
 		iface->if_dev->oper_state = oper_state;
 	}

